3-UsingAPIs/Week2/homework/ex1-programmerFun/index.js

```diff
@@ -17,29 +17,31 @@ Full description at: https://github.com/HackYourFuture/Homework/blob/main/3-Usin
    url with `.shx`. There is no server at the modified url, therefore this 
    should result in a network (DNS) error.
 ------------------------------------------------------------------------------*/
-function requestData(url) {
-  // TODO return a promise using `fetch()`
+async function requestData(url) {
+  const response = await fetch(url);
+  if (!response.ok) {
+    throw 'network errors';
+  } else {
+    return response.json();
+  }
```

Comment:
Best practice is to report errors by creating an Error object. 

Also, both `return` and `throw` cause the function to terminate immediately. No need for an `else` block.

```suggestion
  if (response.ok) {
    return response.json();
  }
  throw new Error('HTTP error');
```